ibis/backends/snowflake/tests/test_datatypes.py

```diff
@@ -0,0 +1,158 @@
+from __future__ import annotations
+
+import pytest
+from pytest import param
+
+import ibis
+import ibis.expr.datatypes as dt
+from ibis.backends.snowflake.datatypes import parse
+from ibis.backends.snowflake.tests.conftest import _get_url
+from ibis.util import gen_name
+
+dtypes = [
+    ("FIXED", dt.int64),
+    ("REAL", dt.float64),
+    ("TEXT", dt.string),
+    ("DATE", dt.date),
+    ("TIMESTAMP", dt.Timestamp(scale=9)),
+    ("VARIANT", dt.json),
+    ("TIMESTAMP_LTZ", dt.Timestamp(scale=9)),
+    ("TIMESTAMP_TZ", dt.Timestamp(timezone="UTC", scale=9)),
+    ("TIMESTAMP_NTZ", dt.Timestamp(scale=9)),
+    ("OBJECT", dt.Map(dt.string, dt.json)),
+    ("ARRAY", dt.Array(dt.json)),
+    ("BINARY", dt.binary),
+    ("TIME", dt.time),
+    ("BOOLEAN", dt.boolean),
+]
+
+
+@pytest.mark.parametrize(
+    ("snowflake_type", "ibis_type"),
+    [
+        param(snowflake_type, ibis_type, id=snowflake_type)
+        for snowflake_type, ibis_type in dtypes
+    ],
+)
+def test_parse(snowflake_type, ibis_type):
+    assert parse(snowflake_type.upper()) == ibis_type
+
+
+@pytest.fixture(scope="module")
+def con():
+    return ibis.connect(_get_url())
+
+
+user_dtypes = [
+    ("NUMBER", dt.int64),
+    ("DECIMAL", dt.int64),
+    ("NUMERIC", dt.int64),
+    ("NUMBER(5)", dt.int64),
+    ("DECIMAL(5, 2)", dt.Decimal(5, 2)),
+    ("NUMERIC(21, 17)", dt.Decimal(21, 17)),
+    ("INT", dt.int64),
+    ("INTEGER", dt.int64),
+    ("BIGINT", dt.int64),
+    ("SMALLINT", dt.int64),
+    ("TINYINT", dt.int64),
```

Comment:
Shoudln't we parse integer types with the right bitwidth here?

Reply:
No, because that's not snowflake's behavior.